### ISSUE TITLE
Tab completion at beginning of command doesn't always work

### ISSUE THREAD

**Issue by asmeurer** [scopatz/xonsh#777]
In the prompt toolkit shell. If I type

```
emacsclient
```

Then move my cursor to the beginning of the line and type `~/a<TAB>`, it completes. But if I type

```
emacsclient -nw
```

and move my cursor to the beginning of the line and type the same thing, it wants to complete 

```
                                                           $(                                      &
                    l                                      $[                                      &=
                   !!                                      ${                                      **                                      >
                   !=                                      %                                       **=
                   !n                                      %=                                      *=
```

**Comment by gforsyth** [comment]
Hey @asmeurer , can you try out #781?  I think I've fixed this behavior.

**Comment by melund** [comment]
I think this was fixed by #781.  @asmeurer I hope it is OK if I close
